packages/loader/driver-utils/src/networkUtils.ts

```diff
@@ -9,24 +9,34 @@ import { ITelemetryLoggerExt } from "@fluidframework/telemetry-utils/internal";
 import { OnlineStatus, canRetryOnError, isOnline } from "./network.js";
 
 /**
+ * Logs network failure details to telemetry.
  * @internal
  */
 export function logNetworkFailure(
 	logger: ITelemetryLoggerExt,
 	event: ITelemetryErrorEventExt,
-	error?: any,
-) {
+	error?: unknown,
+): void {
 	const newEvent = { ...event };
 
-	const errorOnlineProp = error?.online;
+	// Extract online property if present
+	const errorOnlineProp = (error as { online?: string })?.online;
 	newEvent.online =
 		typeof errorOnlineProp === "string" ? errorOnlineProp : OnlineStatus[isOnline()];
 
-	if (typeof navigator === "object" && navigator !== null) {
-		const nav = navigator as any;
-		const connection = nav.connection ?? nav.mozConnection ?? nav.webkitConnection;
-		if (connection !== null && typeof connection === "object") {
-			newEvent.connectionType = connection.type;
+	if (typeof navigator === "object") {
+		// Navigator connection info if available
+		const nav = navigator as unknown as {
+			connection?: unknown;
+			mozConnection?: unknown;
+			webkitConnection?: unknown;
+		};
+		const connObj = nav.connection ?? nav.mozConnection ?? nav.webkitConnection;
+		if (connObj !== undefined && typeof connObj === "object") {
+			const { type } = connObj as { type?: unknown };
+			if (typeof type === "string") {
+				newEvent.connectionType = type;
+			}
```

Comment:
Nit: general guidance is to avoid unnecessary abbreviations where possible for developer accessibility.
```suggestion
		const navigatorWithConnectionProps = navigator as unknown as {
			connection?: unknown;
			mozConnection?: unknown;
			webkitConnection?: unknown;
		};
		const connectionObject = nav.connection ?? nav.mozConnection ?? nav.webkitConnection;
		if (connectionObject !== undefined && typeof connectionObject === "object") {
			const { type } = connectionObjectas { type?: unknown };
			if (typeof type === "string") {
				newEvent.connectionType = type;
			}
```